### PR TITLE
New Custom Massdrop CTRL keymap

### PR DESCRIPTION
**Pull request by rallyrabbit**
New keymap
Adding unicode characters to a second layer
ANSI keymap relatively untouched
Added 4 RGB modes not previously with Massdrop CTRL

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Massdrop Ctrl RallyRabbit
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/README.md:1]: This file should be renamed to lowercase.

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/config.h [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+/*#define NO_ACTION_LAYER*/
+/*#define NO_ACTION_TAPPING*/
+/*#define NO_ACTION_ONESHOT*/
+
+#define RGB_MATRIX_KEYPRESSES
+#define RGB_MATRIX_LED_PROCESS_LIMIT 15
+#define RGB_MATRIX_LED_FLUSH_LIMIT 10
+// RGB Matrix Animation modes. Explicitly enabled
+// For full list of effects, see:
+// https://docs.qmk.fm/#/feature_rgb_matrix?id=rgb-matrix-effects
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/config.h:129]:
```suggestion
// Copyright 2023 @rallyrabbit 
// SPDX-License-Identifier: GPL-2.0-or-later
#pragma once
```

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/keymap.c [hunk 3]
```diff
@@ -0,0 +1,207 @@
+#include QMK_KEYBOARD_H
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/keymap.c:1]: This file is missing a license header.

File: keyboards/massdrop/ctrl/rules.mk [hunk 4]
```diff
@@ -23,7 +23,12 @@ AUDIO_ENABLE = no           # Audio output
...
 AUTO_SHIFT_ENABLE = no      # Auto Shift
+UNICODE_ENABLE = yes        # Enable Unicode Characters/Strings
 
 # Custom RGB matrix handling
 RGB_MATRIX_ENABLE = yes
 RGB_MATRIX_DRIVER = custom
+
+#Custom RGB
+RGB_MATRIX_CUSTOM_USER = yes
+
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/rules.mk:34]:
```suggestion

# Custom RGB matrix handling
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = custom
```

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/rules.mk [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+AUTO_SHIFT_ENABLE = no      # Auto Shift
+UNICODE_ENABLE = yes        # Enable Unicode Characters/Strings
+
+# Custom RGB matrix handling
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = custom
+
+#Custom RGB
+RGB_MATRIX_CUSTOM_USER = yes
+
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/rules.mk:34]:
```suggestion
UNICODE_ENABLE = yes
RGB_MATRIX_CUSTOM_USER = yes
```

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/rgb_matrix_user.inc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+RGB_MATRIX_EFFECT(BAND_SAT_DEEP_RIVER_REVERSE)
+
+/* Step 2. */
+/* Define effects inside the `RGB_MATRIX_CUSTOM_EFFECT_IMPLS` ifdef block */
+#ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+
+/*
+ * Blooming Flower was taken from HorrorTroll
+ */
+#include "flower_blooming/flower_blooming.c"
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/rgb_matrix_user.inc:37]: C files should not be `#include`'d. Either add the code in this file or use a more appropriate way of including it.

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+* Added Flower Blooming of HorrorTroll
+* Added Deep River (unchangeable blue with green saturation that moves)
+* Added Deep River Reverse (unchangeable green with blue saturation that moves)
+
+## To Use
+
+Follow the QMK directions isntalling QMK MSYS.
+* Download the QMK firmware (this repository)
+* Copy config.h and rules.mk to the keyboards/massdrop/ctrl directory
+* Build with "qmk compile -kb massdrop/ctrl -km rallyrabbit"
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/README.md:27]:
```suggestion
* Build with `qmk compile -kb massdrop/ctrl -km rallyrabbit`
```

File: keyboards/massdrop/ctrl/keymaps/rallyrabbit/flower_blooming/flower_blooming.h [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (waffle87) [inline keyboards/massdrop/ctrl/keymaps/rallyrabbit/flower_blooming/flower_blooming.h:1]: This file is missing a license header.

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by rallyrabbit** [comment]
@tzarc if this is on hold to facilitate HorrorTroll's changes into a central location rather than copy pasting everywhere, I can certainly pull those out for now and re-work once that work is complete.

**Comment by rallyrabbit** [comment]
> On hold; awaiting PR from @HorrorTroll to move RGB effects to core instead of copypasting everywhere.

Blooming Flower removed from PR

**Comment by zvecr** [comment]
Unfortunately since this PRs creation, QMK has [changed its stance](https://docs.qmk.fm/#/ChangeLog/20230827?id=notable-changes) on accepting user keymaps to the repository.

> qmk/qmk_firmware is no longer accepting PRs for keymaps other than for manufacturer-supported keymaps. User keymap workflow has been documented [here](https://docs.qmk.fm/#/newbs) for several years. This change is to progressively reduce the maintenance burden on the project, and to allow us to focus on the core features of QMK.

In the near future we will be providing a more user focused workflow for keymaps, but till then the suggestion is to keep changes in your own fork.

While we appreciate the time that was taken on this PR, hopefully you understand we are closing it, so that QMK can improve in the long term.

**Pull request closed without merging**